File: src/commands/greet.ts
import { Command } from "commander";
import { logger } from "../lib/logger.js";

export function greetCommand(): Command {
  return new Command("greet")
    .description("Greet a user")
    .argument("<name>", "Name to greet")
    .option("-u, --uppercase", "Print name in uppercase")
    .action((name: string, opts: { uppercase?: boolean }) => {
      const displayName = opts.uppercase ? name.toUpperCase() : name;
      logger.success(`Hello, ${displayName}!`);
    });
}

File: internal/commands/greet.go
package commands

import (
	"strings"

	"github.com/abisekhsubedi/mycli/internal/logger"
	"github.com/spf13/cobra"
)

func NewGreetCmd() *cobra.Command {
	var uppercase bool

	cmd := &cobra.Command{
		Use:   "greet <name>",
		Short: "Greet a user",
		Args:  cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			name := args[0]
			if uppercase {
				name = strings.ToUpper(name)
			}
			logger.Success("Hello, %s!", name)
			return nil
		},
	}

	cmd.Flags().BoolVarP(&uppercase, "uppercase", "u", false, "Print name in uppercase")

	return cmd
}
